test/exe/validate_extension_manifest_test.cc

```diff
@@ -0,0 +1,75 @@
+#include "envoy/registry/registry.h"
```

Comment:
I don't think it was possible before https://github.com/envoyproxy/envoy/pull/21413. There was no way to query Envoy binary for mapping from extension names to type URLs. I can try rewrite this in a python test, but it seemed to me like a bazel test is a better fit.

Reply:
wait - this is the other manifest file - but i think that is also already being validated - and either way i think it _should_ be validated as part of the python code checks

Reply:
see here for existing extensions checks https://github.com/envoyproxy/pytooling/blob/main/envoy.code.check/envoy/code/check/abstract/extensions.py

Reply:
theoretically - but im struggling to understand why they need to to validate static data - maybe im not understanding tho and this is validating something else

Reply:
@phlax so inside Envoy extensions define their names and type URLs as part of the headers in the code. Extension authors are supposed to put the same name in the manifest (they never had to write type URL anywhere). This new test just ensures that when names match, the type URLs also match in the static manifest and what's defined inside Envoy. That way we can treat the manifest as authoritative and not rely on authors not messing it up.

Reply:
so @kyessenov one small piece of pushback - maybe we could rename the test that was confusting me to a name that doesnt sound like its doing static config/schema validation

not sure what is best - but i already struggle around this with `config_validation` so that is part of my reason for pushing back

maybe just something like `validate_extensions_are_manifest` (thinking out load)

Reply:
thanks! 8)